assets/js/components/flows/FlowsWorkspace.jsx

```diff
@@ -209,115 +212,114 @@ export default ({
+            (selectedNodeId &&
+              selectedNodeId.split(/-(.+)/)[0].replace("-", ""))
           }
-          debug={
-            !!(
-              selectedNodeId &&
-              ["device", "label"].includes(
-                selectedNodeId.split(/-(.+)/)[0].replace("-", "")
-              )
-            )
-          }
-        >
-          <NodeInfo
-            id={
-              (selectedNodeId && selectedNodeId.slice(0, 4) === "edge" && selectedNodeId) ||
-              (selectedNodeId && selectedNodeId.split(/-(.+)/)[1].split("_copy")[0])
-            }
-            type={
-              (selectedNodeId && selectedNodeId.slice(0, 4) === "edge" && "edge") ||
-              (selectedNodeId && selectedNodeId.split(/-(.+)/)[0].replace("-", ""))
-            }
-            elementsMap={elementsMap}
-            onAdrUpdate={onAdrUpdate}
-            onMultiBuyUpdate={onMultiBuyUpdate}
-            onAlertUpdate={onAlertUpdate}
-            hasConnectedEdges={
-              Object.values(elementsMap)
-              .filter(el =>
+          elementsMap={elementsMap}
+          onAdrUpdate={onAdrUpdate}
+          onMultiBuyUpdate={onMultiBuyUpdate}
+          onAlertUpdate={onAlertUpdate}
+          hasConnectedEdges={
+            Object.values(elementsMap).filter(
+              (el) =>
                 isEdge(el) &&
-                (el.source === selectedNodeId ||
-                  el.target === selectedNodeId)
-              ).length > 0
-            }
-            deleteNode={() => {
-              let edges = []
+                (el.source === selectedNodeId || el.target === selectedNodeId)
+            ).length > 0
+          }
+          deleteNode={() => {
+            let edges = [];
 
-              if (selectedNodeId.slice(0, 4) === "edge") {
-                edges = Object.values(elementsMap)
-                  .filter(
-                    (el) =>
-                      isEdge(el) &&
-                      (el.source === selectedNodeId ||
-                        el.target === selectedNodeId)
-                  )
-                  .map((el) => el.id);
-              }
+            if (selectedNodeId.slice(0, 4) === "edge") {
+              edges = Object.values(elementsMap)
+                .filter(
+                  (el) =>
+                    isEdge(el) &&
+                    (el.source === selectedNodeId ||
+                      el.target === selectedNodeId)
+                )
+                .map((el) => el.id);
+            }
 
-              setElements((elsMap) =>
-                omit(elsMap, edges.concat(selectedNodeId))
-              );
+            setElements((elsMap) => omit(elsMap, edges.concat(selectedNodeId)));
 
-              setChangesState(true);
-              setShowInfoSidebar(false);
-            }}
-          />
-        </InfoSidebar>
-      </ReactFlowProvider>
+            setChangesState(true);
+            setShowInfoSidebar(false);
+            setSelectedElements([]);
```

Comment:
main addition here, besides moving the provider to wrap `FlowsWorkspace` in the parent component